packages/contracts/contracts/mixins/MixinSetStateWithAction.sol

```diff
@@ -0,0 +1,153 @@
+pragma solidity 0.4.25;
+pragma experimental "ABIEncoderV2";
+
+import "../libs/LibStateChannelApp.sol";
+import "../libs/LibSignature.sol";
+import "../libs/LibStaticCall.sol";
+
+import "./MAppRegistryCore.sol";
+import "./MAppCaller.sol";
+
+
+contract MixinSetStateWithAction is
+  LibSignature,
+  LibStateChannelApp,
+  MAppRegistryCore,
+  MAppCaller
+{
+
+  struct SignedStateUpdate {
+    // NOTE: We include the full bytes of the state update,
+    //       not just the hash of it as in MixinSetState.
+    bytes encodedState;
+    uint256 nonce;
+    uint256 timeout;
+    bytes signatures;
+  }
+
+  struct SignedAction {
+    bytes encodedAction;
+    bytes signature;
+    bool checkForTerminal;
+  }
+
+  /// @notice Create a dispute regarding the latest signed state and immediately after,
+  /// performs a unilateral action to update it.
+  // @param app An `App` struct specifying the application logic
+  // TODO: Docs
+  /// @dev Note this function is only callable when the state channel is in an ON state
```

Comment:
Is there a modifier enforcing this?

Reply:
Not a modifier but a require statement

https://github.com/counterfactual/monorepo/blob/9dad2a7f4b39f678f7b579734220e1bf31c2928d/packages/contracts/contracts/mixins/MixinSetStateWithAction.sol#L57-L60